QModManager/Utility/Config.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace QModManager.Utility
+{
+    using Oculus.Newtonsoft.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+
+    internal static class Config
+    {
+        internal static class FIELDS
+        {
+            internal const string CHECK_FOR_UPDATES = "Check for updates";
+            internal const string ENABLE_CONSOLE = "Enable console";
+            internal const string ENABLE_DEBUG_LOGS = "Enable debug logs";
+            internal const string ENABLE_DEV_MODE = "Enable developer mode";
+        }
```

Comment:
I don't know about this... @PrimeSonic?

Reply:
Rather than expose constants and have to pass them in as parameters, instead, define an `interface` for the config and in there you can make specific `get` properties with the correct type and all the internal handling you'll need.

An example of this in action is in MSU
[IModConfig](https://github.com/PrimeSonic/PrimeSonicSubnauticaMods/blob/master/MoreCyclopsUpgrades/Config/IModConfig.cs])  
```CS
// The use of the interface keeps external call to the the config class clean
internal class ModConfig : IModConfig
{
    private static readonly ModConfig main = new ModConfig();

    // The singleton pattern allows you to use the interface from a static context
    internal static IModConfig Main
    {
        get
        {
            // If we need to load values from a file,
            // that can be done on demand
            if (!main.initialized)
                main.Initialize();

            return main;
        }
    }
```

Reply:
I'd personally like to keep the Config class static rather than make it a singleton

Reply:
Static classes can't implement interfaces.  
So with that in mind, your next best option is to define the static methods sort of like this:
```CS
internal static bool CheckForUpdates { get; private set; }
internal static bool EnableConsole { get; private set; }
internal static bool EnableDebugLogs { get; private set; }
internal static bool EnableDevMode { get; private set; }
```
You can (and probably should) hide away all of the logic around reading the config file from the rest of the assembly by keeping those methods private, and exposing only what external consumers actually care about: the final value.

Reply:
You will probably end up making these as "get only" properties which contain the necessary code to read from the config file.